coin_chk_libhdr.m4

```diff
@@ -0,0 +1,432 @@
+# run_autotools.
+
+# We're trying to control two distinct things: should the library be used
+# by default (yes or no), and should the link check use only the function
+# (sep(arate) or both the includes and the function (tog(ether)). This macro
+# looks at the possibilities and returns the appropriate string. [which]
+# ($1) controls whether it returns library usage ('usage') or the form
+# of the link check ('link'). [dfltaction] ($2) is the parameter to
+# be analysed. [dfltuse] ($3) specifies the default for usage, in case
+# [dfltaction] doesn't specify it. [dfltlink] ($4) covers the default for
+# the link check, in case [dfltaction] doesn't specify it.
+
+# It's critical that the macro produce a result with no leading or trailing
+# spaces! Be careful if you modify it. In normal use, you want this macro to
+# expand immediately, so don't quote it.
+
+AC_DEFUN([AC_COIN_LIBHDR_DFLT_HACK],
+[m4_case([$2],
+   [yessep],[m4_if([$1],[link],[sep],[yes])],
+   [yestog],[m4_if([$1],[link],[tog],[yes])],
+   [nosep],[m4_if([$1],[link],[sep],[no])],
+   [notog],[m4_if([$1],[link],[tog],[no])],
+   [yes],[m4_if([$1],[link],[$4],[yes])],
+   [no],[m4_if([$1],[link],[$4],[no])],
+   [sep],[m4_if([$1],[link],[sep],[$3])],
+   [tog],[m4_if([$1],[link],[tog],[$3])],
+   [m4_if([$1],[link],[$4],[$3])])])dnl 	# COIN_LIBHDR_DFLT_HACK
+  
+
+
+###########################################################################
+#                         FIND_PRIM_LIBHDR                                #
+###########################################################################
+
+# COIN_FIND_PRIM_LIBHDR([prim],[lflgs],[cflgs],[dflgs],
+#                       [function-body],[includes],
+#                       [dfltaction],[cmdlineopts])
+
+# Determine whether we can use primary library prim ($1) and assemble
+# information on the required linker flags (prim_lflags), compiler flags
+# (prim_cflags), and data directories (prim_data) as specified by cmdlineopts
+# ($8).
+
+# A compile check will be performed if [includes] ($6) is specified.  A link
+# check will be performed if [function-body] ($5) is specified. The default is
+# to use a program composed by concatenating [includes] and [function-body]
+# with link flags specified by [lflgs] ($2), and compiler flags specified by
+# [cflgs] ($3). If [dfltaction] ($7) ends in 'sep', the link check will be
+# performed without the header (see LIBHDR_DFLT_HACK).
+
+# There's really no reasonable assumptions that can be made about how a data
+# directory is specified, so there's no attempt to test for existence.
+
+# [cmdlineopts] ($8) specifies the set of configure command line options
+# processed: 'nodata' processes --with-prim, --with-prim-lflags,
+# and --with-prim-cflags; 'dataonly' processes only --with-prim and
+# --with-prim-data; anything else ('all' works well) processes all four
+# command line options. Shell code produced by the macro is tailored based
+# on [cmdlineopts]. `nodata' is the default.
```

Comment:
Would it make sense to split `COIN_FIND_PRIM_LIBHDR` into two macros, one that checks lib and header, and one that checks the datadir? That should allow to get rid of the `cmdlineopts` argument, but I'm not sure how much of `COIN_FIND_PRIM_LIBHDR` would need to be duplicated.

Reply:
Could be done --- `FIND_PRIM_LIBHDR` would separate fairly cleanly --- but it just pushes the complexity back into `CHK_LIBHDR` unless we create a separate `CHK_DATA` front end.  I guess I don't see a real advantage.